pkg/adapter/nimbus-kyverno/processor/kcpbuilder.go

```diff
@@ -39,60 +41,89 @@ func BuildKcpsFrom(logger logr.Logger, cnp *v1.ClusterNimbusPolicy) []kyvernov1.
 }
 
 // buildKpFor builds a KyvernoPolicy based on intent ID supported by Kyverno Policy Engine.
-func buildKcpFor(id string, cnp *v1.ClusterNimbusPolicy) kyvernov1.ClusterPolicy {
+func buildKcpFor(id string, cnp *v1.ClusterNimbusPolicy, rule v1.Rule) kyvernov1.ClusterPolicy {
```

Comment:
Why are you passing v1.Rule again, when the rule is already available as part of v1.ClusterNimbusPolicy ?